File: 345_Reverse_Vowels_of_a_String/solution.py
class Solution:
    def reverseVowels(self, s: str) -> str:
        vowels, has = "aeiouAEIOU", []
        for char in s:
            if char in vowels:
                has.append(char)

        has.reverse()
        s, index = list(s), 0
        for i in range(len(s)):
            if s[i] in vowels:
                s[i] = has[index]
                index += 1

        return "".join(s)

File: 345_Reverse_Vowels_of_a_String/solution.c
bool isvowel(char c) { return c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u' || c == 'A' || c == 'E' || c == 'I' || c == 'O' || c == 'U'; }

char* reverseVowels(char* s) {
    int i = 0, size = 0;
    while (s[i] != '\0') { if (isvowel(s[i])) size++; i++; }
    i = 0;

    char* vowels = (char*)malloc(size * sizeof(char));
    int index = 0;
    while (s[i] != '\0') { if (isvowel(s[i])) vowels[index++] = s[i]; i++; }
    i = 0;

    int n = strlen(s);
    while (s[i] != '\0') { if (isvowel(s[i])) { s[i] = vowels[size - 1]; size--; } i++; }

    free(vowels);
    return s;
}
